ruptime

```diff
@@ -49,153 +62,163 @@ fi
+			which free >/dev/null && mem=$(free -b |grep ^Mem: |awk '{printf "%.0f\n", $2/1024/1024/1024}')
+			which free >/dev/null || mem=$(sysctl -n hw.memsize|awk '{printf "%.0f\n", $1/1024/1024/1024}')
+			cores=$(getconf _NPROCESSORS_ONLN)
+			gpu=0; which nvidia-smi >/dev/null && gpu=$(nvidia-smi -L | wc -l)
+			# macOS number-of-GPU-cores system_profiler SPDisplaysDataType |grep "Cores"
+			u=$(printf "%-28s %s\n" "$k" "$biosdate $sys $cores $mem $gpu" | encrypt)
+			# macOS system_profiler SPHardwareDataType
+			;;
+		rac)
+			ac=$(ac | awk '{print $2}')
+			nu=$(ac -p |sed 1d |wc -l)
+			u=$(printf "%-28s %s %s\n" "$k" "$ac" "$nu" | encrypt)
+			;;
+		rwho)
+			w=$(who |sed 's/ [a-zA-Z:]/ '"$k"':&/1;s,: ,:,1')
+			wl=$(echo -n "$w"|wc -l)
+			if [ "${wl}" -ne 0 ]; then
+				u=$(printf "%s\n" "$w" | encrypt)
+			else
+				exit 0
+			fi
+			;;
+		rbench)
+			dt=$(which date)
+			which gdate >/dev/null && dt=$(which gdate)
+			which free >/dev/null && m=$(free -b |grep '^Mem:' | awk '{printf "%d\n", $2/1024^3}')
+			which free >/dev/null || m=$(sysctl -n hw.memsize| awk '{printf "%d\n", $1/1024^3}')
+			start=$($dt +%s.%N)
+			memtester 16M 1 &>/dev/null
+			end=$($dt +%s.%N)
+			mem=$(echo "$end $start $m"| awk '{printf "MEM %.2f %s GB\n", $1-$2, $3}')
+			cores=$(getconf _NPROCESSORS_ONLN)
+			start2=$($dt +%s.%N)
+			echo 'scale=3000;4*a(1)' | bc -l &>/dev/null
+			end2=$($dt +%s.%N)
+			cpu=$(echo "$end2 $start2 $cores"| awk '{printf "CPU %.2f %s\n", $1-$2, $3}')
+			u=$(printf "%-28s %s %s\n" "$k" "$mem" "$cpu" | encrypt)
+			;;
+		rnet)
+			netdev=$(ip route show to default |head -n1 | cut -d\  -f5)
+			s="$netdev $(cat "/sys/class/net/${netdev}/speed" 2>/dev/null)Mb/s"
+			if iwconfig "$netdev" &>/dev/null; then
+				l=$(iwconfig 2>/dev/null|grep "Link Quality=" | sed "s,Link Quality=,," | awk '{print "scale=2; "$1"*100"}' | bc -l)
+				s2=$(iwconfig "$s" 2>/dev/null|grep "Bit Rate" | sed "s,Bit Rate=,,;s, Mb,Mb,"|awk '{print $1}')
+				u=$(printf "%-28s %s %s %.0f\n" "$k" "$s" "$s2" "$l" | encrypt)
+			else
+				netdev=$(ip r |grep ^default |head -1|sed "s,.*dev ,,g" |awk '{print $1}'|head -1)
+				s="$netdev  $(cat "/sys/class/net/${netdev}/speed" 2>/dev/null)Mb/s"
+				u=$(printf "%-28s %s\n" "$k" "$s" | encrypt)
+			fi
+			# macOS
+			# scutil --nwi |tail -1|awk '{print $NF}'
+			# /System/Library/PrivateFrameworks/Apple80211.framework/Versions/Current/Resources/airport -I
+			;;
+		rdisk)
+			shopt -s nullglob
+			s="$(for a in /dev/ada? /dev/sd? /dev/xvd? /dev/vd? /dev/nvme?n1 /dev/mmcblk? /dev/md[0-9]* ; do
+				size=
+				if fdisk -l "$a" >/dev/null 2>&1; then
+					size="$(fdisk -l "$a" |grep '^Disk' | awk '{printf "%1.0f\n", $5/1000/1000/1000;exit}')"
+					echo -n "${a#/dev/} $size "
+				fi
+			done)"
+			shopt -u nullglob
+			u=$(printf "%-28s %s\n" "$k" "$s" | encrypt)
+			# macOS: diskutil info /dev/disk0 |grep Size
+			;;
```

Comment:
that part is just ...AAAAAHHHHH!!!

This might be an alternative that will be much safer..  AND WORKS w/o UID=0.....
```
% lsblk -pd -o NAME,SIZE -e "$(awk '/(zram|zvol)/ {printf "%s,",$1}' /proc/devices)"7,11
/dev/sda 0B
/dev/nvme1n1 953.9G
/dev/nvme0n1 476.9G
# removes faulty stuff like sd-card slots...
% lsblk -pd -o NAME,SIZE -e "$(awk '/(zram|zvol)/ {printf "%s,",$1}' /proc/devices)"7,11 --noempty
/dev/nvme1n1 953.9G
/dev/nvme0n1 476.9G
# well i guess this one 
% lsblk -nd -o NAME,SIZE -e "$(awk '/(zram|zvol)/ {printf "%s,",$1}' /proc/devices)"7,11 --noempty
nvme1n1 953.9G
nvme0n1 476.9G
% lsblk -nd -o NAME,SIZE -e "$(awk '/(zram|zvol)/ {printf "%s,",$1}' /proc/devices)"7,11 --noempty  |tr '\n' ' '
nvme1n1 953.9G nvme0n1 476.9G %

```